wait_for/__init__.py

```diff
@@ -64,7 +64,12 @@ def _get_context(func, message=None):
         filename = f_code.co_filename
         if not message:
             if is_lambda_function(func):
-                message = 'lambda defined as `{}`'.format(inspect.getsource(func).strip())
+                try:
+                    message = 'lambda defined as `{}`'.format(inspect.getsource(func).strip())
+                except IOError as ioerror:
+                    # We are probably in interactive python shell or debugger,
+                    # we cannot get source of the lambda.
+                    message = "lambda (and couldn't get it's source code)".format(ioerror)
```

Comment:
I know we _can_ detect if we are in an ipython/jupyter - perhaps instead of saying we couldn't get it, we could mention that it may be because we are in an interactive shell?

Reply:
I added some info in the message.